plugins/hydra_ray_launcher/hydra_plugins/hydra_ray_launcher/conf/__init__.py

```diff
@@ -0,0 +1,213 @@
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import Any, Dict, List, Optional
+
+from hydra.core.config_store import ConfigStore
+from hydra.types import ObjectConf
+from omegaconf import MISSING
+
+
+@dataclass
+class RayInitConf(Dict[str, Any]):
+    """ parameters for ray.init()"""
+
+    address: Optional[str] = MISSING
+    num_cpus: Optional[int] = MISSING
+    num_gpus: Optional[int] = MISSING
+
+
+@dataclass
+class RayRemoteConf(Dict[str, Any]):
+    """ parameters for @ray.remote() """
+
+    num_cpus: Optional[int] = 1
+    num_gpus: Optional[int] = 0
+
+
+@dataclass
+class RayConf:
+    ray_init_cfg: RayInitConf = MISSING
+    ray_remote_cfg: RayRemoteConf = field(default_factory=RayRemoteConf)
+
+
+@dataclass
+class RayLocalLauncherConf(ObjectConf):
+    params: RayConf = RayConf()
+
+
+# Ray AWS config
+
+
+class RayAutoScalingMode(Enum):
+    default = "default"
+    aggressive = "aggressive"
+
+
+@dataclass
+class RayDockerConf(Dict[str, Any]):
+    """
+    This executes all commands on all nodes in the docker container,
+    and opens all the necessary ports to support the Ray cluster.
+    Empty string means disabled.
+
+    """
+
+    # e.g., tensorflow/tensorflow:1.5.0-py3
+    image: str = ""
+    # e.g. ray_docker
+    container_name: str = ""
+    # If true, pulls latest version of image. Otherwise, `docker run` will only pull the image
+    # if no cached version is present.
+    pull_before_run: bool = True
+    # Extra options to pass into "docker run"
+    run_options: List[str] = field(default_factory=list)
+
+
+@dataclass
+class RayProviderConf(Dict[str, Any]):
+    type: str = "aws"
+    region: str = "us-west-2"
+    availability_zone: str = "us-west-2a,us-west-2b"
+    cache_stopped_nodes: bool = False
+
+
+@dataclass
+class RayAuthConf(Dict[str, Any]):
+    ssh_user: str = "ubuntu"
+
+
+@dataclass
+class RayNodeConf(Dict[str, Any]):
+    """
+    Additional options in boto docs.
+    https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2.html
+    """
+
+    InstanceType: str = "m5.large"
+    ImageId: str = "ami-008d8ed4bd7dc2485"
+
+
+@dataclass
+class RsyncConf:
+    """
+    If your code contains more than one modules/files, you need to sync your code to the remote cluster.
+    The code will be synced to a tmp dir on the cluster.
+    code_dir could be a relative dir to where you run your app or a absolute dir.
+    Leave this empty if you don't need to sync code.
+
+    The conf is equivalent to run:
+    rsync {source} {target} --include={include} --exclude={exclude}
+
+    source/target will either be remote/local or local/remote depending on sync_up or sync down
+    Read RayAWSConf for more details.
+    """
+
+    source_dir: Optional[str] = None
+    target_dir: Optional[str] = None
+    include: List[str] = field(default_factory=list)
+    exclude: List[str] = field(default_factory=list)
+
+
+@dataclass
+class RayClusterConf(Dict[str, Any]):
```

Comment:
Why are you extending Dict?
Are there any parameters that cannot be specified in the config itself that you are using?

Reply:
Ive cleaned up all the conf's that doesnt need to extend Dict

Reply:
So why are you extending Dict?

Reply:
i missed this one while cleaning up. will remove.